### PR TITLE
Add more aspects to ReservedStateVersion test

### PR DESCRIPTION
**Pull request by thecoop**
Also refactor ReservedClusterStateServiceTest to combine class definitions

### CODE REVIEW

**Comment by thecoop** [comment]
@elasticmachine update branch

File: server/src/test/java/org/elasticsearch/reservedstate/service/ReservedClusterStateServiceTests.java [hunk 1]
```diff
@@ -450,15 +428,44 @@ public void testCheckMetadataVersion() {
...
         );
-    }
 
-    private ReservedClusterStateHandler<Map<String, Object>> makeHandlerHelper(final String name, final List<String> deps) {
-        return new ReservedClusterStateHandler<>() {
-            @Override
-            public String name() {
-                return name;
-            }
+        ClusterState state = ClusterState.builder(new ClusterName("test")).metadata(Metadata.builder().put(operatorMetadata)).build();
```
Comment (thecoop) [inline server/src/test/java/org/elasticsearch/reservedstate/service/ReservedClusterStateServiceTests.java:432]: These are the additional tests added

Comment (ldematte) [reply server/src/test/java/org/elasticsearch/reservedstate/service/ReservedClusterStateServiceTests.java:432]: It's not clear to me what we are testing here; could you add some comments, or maybe separate this test into a different ones with better names, or add assertion messages so it's clear what we expect and why?

Comment (thecoop) [reply server/src/test/java/org/elasticsearch/reservedstate/service/ReservedClusterStateServiceTests.java:432]: Looking at it, the new checks subsume the old ones - so I've just removed the old checks.

This is basically testing that when the state version or release version is out of date, the cluster state is left as-is

Comment (ldematte) [reply server/src/test/java/org/elasticsearch/reservedstate/service/ReservedClusterStateServiceTests.java:423]: > testing that when the state version or release version is out of date, the cluster state is left as-is

I think that would be great for our future selves to add this as the assertion message.

**Review by ldematte — APPROVED** [review]
One small comment, otherwise LGTM

**Pull request merged**